pkg/pipeline/extract/conntrack/store.go

```diff
@@ -100,10 +100,11 @@ func (cs *connectionStore) expireConnection(hashId uint64) {
 	}
 	groupIdx := cs.hashId2groupIdx[hashId]
 	mom := cs.groups[groupIdx].mom
-	// Set the expiry time to the zero value of Time
-	conn.setExpiryTime(time.Time{})
-	// Move to the front of the list
-	err := mom.MoveToFront(utils.Key(hashId), expiryOrder)
+	// Set the expiry time to half of EndConnectionTimeout
+	timeoutMs := cs.groups[groupIdx].scheduling.EndConnectionTimeout.Duration.Milliseconds()
+	conn.setExpiryTime(time.Now().Add(time.Duration(timeoutMs/2) * time.Millisecond))
+	// Move to the back of the list
+	err := mom.MoveToBack(utils.Key(hashId), expiryOrder)
```

Comment:
I suggest to add half of the configured `EndConnectionTimeout` as safeguard

As far as I see in my tests, it's solving the FIN flag issue